c7n/resources/appmesh.py

```diff
@@ -0,0 +1,280 @@
+"""
+AppMesh Communications
+"""
+from botocore.exceptions import ClientError
+
+from c7n import query
+from c7n.exceptions import PolicyExecutionError
+from c7n.manager import resources
+from c7n.query import ChildResourceManager, QueryResourceManager, TypeInfo, ChildDescribeSource
+from c7n.utils import local_session
+
+
+@resources.register('appmesh-mesh')
+class AppmeshMesh(QueryResourceManager):
+    # interior class that defines the aws metadata for resource
+    class resource_type(TypeInfo):
+        service = 'appmesh'
+        arn_type = "mesh"
+
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appmesh-virtualnode.html
+        cfn_type = config_type = 'AWS::AppMesh::Mesh'
+
+        # field in response containing the identifier
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appmesh-mesh.html
+        id = name = 'meshName'
+
+        # enum_spec defines the boto3 call used to find at least basic
+        # details all resources of the relevant type.  the data per
+        # resource can be further enriched by a detail_spec function.
+        # enum_spec is also used when we've received an event in which
+        # case the results from enum_spec are filtered to include only
+        # those in the event.
+        #
+        # If the enum function chosen allows a filter param to be
+        # specified then the filtering can be done on the server
+        # side. For instance, ASG uses "describe_auto_scaling_groups"
+        # as the enum function and "AutoScalingGroupNames" as a filter
+        # param to that function to limit the server side work.
+        # However, it seems that most "cloud custodian" integrations

```

Comment:
enum_spec is for the default describe source, multiple sources are supported per resource, like config, etc. if the enum_spec doesn't support filtering, the intent is that based on cardinality (ie. doesn't matter for cloud trail, but does matter for high cardinality resource) a describe source override is used, which can use the detail spec or the method directly.

Reply:
Cool